generator/feature_segments_checker/feature_segments_checker.cpp

```diff
@@ -75,14 +79,37 @@ void WriteCSV(Cont const & cont, string const & fileName)
     fout << a.first << "," << a.second << endl;
 }
 
-/// \returns expected point altitude in meters according to linear model.
-double ExpectedPointAltitude(int16_t startAltitudeMeters, int16_t endAltitudeMeters,
-                             double distFromStartMeters, double featureLengthMeters)
+/// \returns y = k * x + b. It's an expected altitude in meters.
+double GetY(double k, double b, double d)
 {
-  if (featureLengthMeters == 0.0)
-    return 0;
-  double const k = (endAltitudeMeters - startAltitudeMeters) / featureLengthMeters;
-  return startAltitudeMeters + k * distFromStartMeters;
+  return k * d + b;
+}
+
+/// \brief Calculates factors |k| and |b| of a line using linear least squares method.
+/// \returns false in case of error (e.g. if the line is parallel with ordinate axis)
+/// and true otherwise.
+bool LinearLeastSquaresFactors(vector<double> const & xs, vector<double> const & ys, double & k, double & b)
+{
+  size_t const n = xs.size();
+  long double mx = 0, my = 0, mxy = 0, mx2 = 0;
+  for (int i = 0; i < n; ++i)
+  {
+    mx += xs[i];
+    my += ys[i];
+    mxy += xs[i] * ys[i];
+    mx2 += xs[i] * xs[i];
+  }
+  mx /= n;
```

Comment:
Moreover, there's a variance in measurements, if you wish, try to estimate it. It's quite big, I bet that it's bigger than 0.003 )